analyzers/tests/SonarAnalyzer.UnitTest/Helpers/ParameterLoaderTest.cs

```diff
@@ -213,5 +221,24 @@ private static SonarCompilationReportingContext CreateCompilationWithOption(stri
             var compilationContext = new CompilationAnalysisContext(compilation, options, _ => { }, _ => true, default);
             return new(AnalysisScaffolding.CreateSonarAnalysisContext(), compilationContext);
         }
+        private string GenerateSonarLintWithParametrizedRule(string ruleId, string key, string value)
+        {
+            var ruleParameters = new List<SonarLintXmlRule>()
+            {
+                new SonarLintXmlRule()
+                {
+                    Key = ruleId,
+                    Parameters = new List<SonarLintXmlKeyValuePair>()
+                    {
+                        new SonarLintXmlKeyValuePair()
+                        {
+                            Key = key,
+                            Value = value
+                        }
+                    }
+                }
+            };
+            return AnalysisScaffolding.CreateSonarLintXml(TestContext, rulesParameters: ruleParameters);
+        }
```

Comment:
```suggestion
                new SonarLintXmlRule()
                {
                    Key = ruleId,
                    Parameters = Parameters(key, value)
                }
            };
            return AnalysisScaffolding.CreateSonarLintXml(TestContext, rulesParameters: ruleParameters);

            List<SonarLintXmlKeyValuePair> Parameters(string key, string value) =>
                new()
                {
                new SonarLintXmlKeyValuePair()
                    {
                        Key = key,
                        Value = value
                    }
                };
        }
```

Reply:
I would definitely adopt this local method to populate >1 rule parameters. Since we're only interested in one parameter at a time, I'd keep it more concise (line-wise) and leave it as-is in the current implementation.